cupyx/scipy/sparse/compressed.py

```diff
@@ -1026,40 +1041,54 @@ def _arg_minor_reduce(self, ufunc, axis):
         # Call to the appropriate kernel function
         if axis == 1:
             # Create the vector to hold output
-            value = cupy.zeros(self.shape[0]).astype(cupy.int64)
+            # Note: it's important to set "int" here, following what SciPy
+            # does, as the outcome dtype is platform dependent
+            value = cupy.zeros(self.shape[0], dtype=int)
 
             # Perform the calculation
+            ker_name = '_arg_reduction<{}, {}>'.format(
+                _scalar.get_typename(self.data.dtype),
+                _scalar.get_typename(value.dtype))
             if ufunc == cupy.argmax:
-                self._max_arg_reduction_kern(
-                    (self.shape[0],), (1,),
-                    (self.data.astype(cupy.float64), self.indices,
+                ker_name = 'max' + ker_name
+                ker = self._max_arg_reduction_mod.get_function(ker_name)
+                ker((self.shape[0],), (1,),
+                    (self.data, self.indices,
                      self.indptr[:len(self.indptr) - 1],
                      self.indptr[1:], cupy.int64(self.shape[1]),
                      value))
             if ufunc == cupy.argmin:
-                self._min_arg_reduction_kern(
-                    (self.shape[0],), (1,),
-                    (self.data.astype(cupy.float64), self.indices,
+                ker_name = 'min' + ker_name
+                ker = self._min_arg_reduction_mod.get_function(ker_name)
+                ker((self.shape[0],), (1,),
```

Comment:
dupped calls :), lets take the last one out.

Reply:
Done in c7a03be.